src/protocol/templates/ping1d.h.in

```diff
@@ -0,0 +1,171 @@
+/*
+* ping1d.h
+* A device API for the Blue Robotics Ping1D echosounder
+*
+*~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!
+* THIS IS AN AUTOGENERATED FILE
+* DO NOT EDIT
+*~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!
+*
+*/
+
+#pragma once
+
+#include "Arduino.h"
+#include "Stream.h"
+#include "pingmessage.h"
+#include "pingmessage_all.h"
+#include "ping_parser.h"
+
+class Ping1D
+{
+
+public:
+
+    /** Constructor
+     *  @param ser: The device I/O
+     *  @param baudrate: The device I/O baudrate
+     */
+    Ping1D(Stream& ser, uint32_t baudrate);
```

Comment:
What you think to receive a SerialDevice as argument ?

Reply:
`Ping1D(HardwareSerial& serial)`
Let the user to configure the serial port and baudrate that he want to use.

Reply:
SoftwareSerial and HardwareSerial inherit Stream.

I think I am doing the best thing right now.

> Let the user to configure the serial port and baudrate that he want to use.

yes, this is the idea, I removed stale baudrate parameter.